packages/runtime/container-runtime/src/gc/gcHelpers.ts

```diff
@@ -241,9 +241,10 @@ export function unpackChildNodesGCDetails(gcDetails: IGarbageCollectionDetailsBa
 		}
 
 		assert(id.startsWith("/"), 0x5d9 /* node id should always be an absolute route */);
-		// TODO Why are we non null asserting here
-		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-		const childId = id.split("/")[1]!;
+		const childId = id.split("/")[1];
+		if (childId === undefined) {
```

Comment:
Better to assert, this is impossible.  How about remove assert 0x5d9 2 lines up and just assert here that childId is defined.